king_phisher/client/plugins.py

```diff
@@ -543,3 +552,208 @@ class ClientPluginManager(plugins.PluginManagerBase):
 	_plugin_klass = ClientPlugin
 	def __init__(self, path, application):
 		super(ClientPluginManager, self).__init__(path, (application,))
+
+class RepoCache(object):
+	"""
+	RepoCache is used to hold basic information on repos that is to be cached, or pulled from cache.
+	"""
+	__slots__ = ('_id', 'title', 'collections_types', 'url')
+
+	def __init__(self, _id, title, collections_types, url):
+		self._id = _id
+		self.title = title
+		self.collections_types = collections_types
+		self.url = url
+
+	@property
+	def id(self):
+		return self._id
+
+	@property
+	def collections(self):
+		return self.collections_types
+
+	def to_dict(self):
+		repo_cache_values = {
+			'id': self._id,
+			'title': self.title,
+			'collections': self.collections_types,
+			'url': self.url
+		}
+		return repo_cache_values
+
+class CatalogCache(object):
```

Comment:
Same thing about using a `namedtuple` here. For the repositories, you'd just create that dictionary ahead of time and pass it to the namedtuple.